### PR TITLE
Prevent a prohibited user from logging in (from the login page)

### PR DESCRIPTION
**Pull request by r-52**
right now, a user with the `Prohibit` can still log in which can cause some inconsistency. The user can click on stuff like `Issues` and now the issue's tab tells the user about the prohibit status.

The main problem (at least in my case) is, if you log in as a user with the `Prohibit` flag set to `true`, the Dashboard freezes the whole tab and it takes a long, long time till the website becomes responsive. In my case, it looked like the chrome tab hang up.

With this PR I've changed the regular login process to check the `Prohibit` flag and prevent the login right at the login process. This prevents the user from using the site.

It's just an idea to protect the instance at the login-level. 

I'm not sure if it's the right way to include a new translation?

### CODE REVIEW

File: models/login_source.go [hunk 1]
```diff
@@ -632,6 +632,9 @@ func UserSignIn(username, password string) (*User, error) {
 		switch user.LoginType {
 		case LoginNoType, LoginPlain, LoginOAuth2:
 			if user.ValidatePassword(password) {
+				if user.ProhibitLogin {
```
Comment (lunny) [inline models/login_source.go]: Also should check in `ExternalUserLogin`

**Review by lunny — CHANGES_REQUESTED** [review]

File: models/login_source.go (hunk 1 above)
Comment (r-52) [reply models/login_source.go]: @lunny thanks! I've force pushed a change

File: models/error.go [hunk 2]
```diff
@@ -90,6 +90,20 @@ func (err ErrUserNotExist) Error() string {
 	return fmt.Sprintf("user does not exist [uid: %d, name: %s, keyid: %d]", err.UID, err.Name, err.KeyID)
 }
 
+// ErrUserLoginProhibited represents an error during the login if the user is blocked
+type ErrUserLoginProhibited struct {
+	Name string
+}
+
+func IsErrUserLoginProhibited(err error) bool {
```
Comment (jonasfranz) [inline models/error.go:99]: Add comment

File: models/error.go [hunk 3]
```diff
@@ -90,6 +90,20 @@ func (err ErrUserNotExist) Error() string {
...
+	_, ok := err.(ErrUserLoginProhibited)
+	return ok
+}
+
+func (err ErrUserLoginProhibited) Error() string {
```
Comment (jonasfranz) [inline models/error.go:105]: Add comment

**Review by jonasfranz — CHANGES_REQUESTED** [review]

File: models/error.go (hunk 2 above)
Comment (r-52) [reply models/error.go:99]: @JonasFranzDEV done :)

File: models/error.go (hunk 3 above)
Comment (r-52) [reply models/error.go:105]: @JonasFranzDEV done :)

File: models/login_source.go (hunk 1 above)
Comment (lunny) [reply models/login_source.go]: @romankl and could you also add check `IsActive` I think maybe it's missed here.

**Review by lunny — CHANGES_REQUESTED** [review]

**Comment by r-52** [comment]
@lunny I've changed it

**Comment by lafriks** [comment]
@lunny @jonasfranz resolved and fixed tests, please review

**Comment by lunny** [comment]
After send my PR #6106 and just found there is also a PR to do that.

**Comment by lunny** [comment]
Will be replaced by #6106. Thanks @romankl

**Pull request closed without merging**